x/programs/rust/wasmlanche-sdk/src/program.rs

```diff
@@ -6,9 +6,30 @@ use crate::{
     Gas,
 };
 use borsh::{BorshDeserialize, BorshSerialize};
-use std::{cell::RefCell, collections::HashMap};
+use std::{cell::RefCell, collections::HashMap, io::Read};
 use thiserror::Error;
 
+#[cfg_attr(feature = "debug", derive(Debug))]
+pub struct DeferDeserialize(Vec<u8>);
+
+impl DeferDeserialize {
+    /// # Panics
+    /// Panics if there was an issue deserializing the value
+    #[allow(clippy::must_use_candidate)]
+    pub fn deserialize<T: BorshDeserialize>(self) -> T {
+        let Self(bytes) = self;
+        borsh::from_slice(&bytes).expect("failed to deserialize")
+    }
```

Comment:
```suggestion
    pub fn deserialize<T: BorshDeserialize>(self) -> Result<T, borsh::io::Error> {
        borsh::from_slice(&self.0)
    }
```
Don't have to change this to one line, but you should let the user decide what to do in the case of an error.

Reply:
Done, the `deserialize` function now returns a `Result`. Thanks!